plugin/evm/vm_test.go

```diff
@@ -369,6 +370,44 @@ func issueAndAccept(t *testing.T, issuer <-chan engCommon.Message, vm *VM) snowm
 	return blk
 }
 
+func TestPreSubnetEVMUpgrade(t *testing.T) {
+	ctx, dbManager, genesisBytes, issuer := setupGenesis(t, genesisJSONPreSubnetEVM)
+	vm := &VM{}
+	err := vm.Initialize(
+		context.Background(),
+		ctx,
+		dbManager,
+		genesisBytes,
+		[]byte(""),
+		[]byte(""),
+		issuer,
+		[]*engCommon.Fx{},
+		nil,
+	)
+
+	// we expect an error when subnet evm upgrade is nil in chain config
+	require.Error(t, err)
+}
+
+func TestSubnetEVMUpgradeNotEnabledAtGenesis(t *testing.T) {
+	ctx, dbManager, genesisBytes, issuer := setupGenesis(t, genesisJSONSubnetEVMLateEnablement)
```

Comment:
alternatively we can parse the json for `genesisJSONSubnetEVM` and modify the value in the test.